lib/src/main/java/com/cloudinary/android/uploadwidget/ui/ThumbnailsAdapter.java

```diff
@@ -59,16 +61,35 @@ public void onClick(View v) {
         });
 
         Context context = holder.itemView.getContext();
-        float thumbnailSize = context.getResources().getDimension(R.dimen.thumbnail_size);
-        BitmapManager.get().load(context, uri, (int) thumbnailSize, (int) thumbnailSize, new BitmapManager.LoadCallback() {
-            @Override
-            public void onSuccess(Bitmap bitmap, Dimensions originalDimensions) {
-                holder.imageView.setImageBitmap(bitmap);
-            }
-
-            @Override
-            public void onFailure() { }
-        });
+        int thumbnailSize = (int) context.getResources().getDimension(R.dimen.thumbnail_size);
+
+        MediaType mediaType = UriUtils.getMediaType(context, uri);
+        if (mediaType == MediaType.IMAGE) {
+            holder.mediaTypeIcon.setVisibility(View.GONE);
+
+            BitmapManager.get().load(context, uri, thumbnailSize, thumbnailSize, new BitmapManager.LoadCallback() {
+                @Override
+                public void onSuccess(Bitmap bitmap, Dimensions originalDimensions) {
+                    holder.imageView.setImageBitmap(bitmap);
```

Comment:
This can cause some issues if the viewholder is already bound to another image (e.g. when the user fast-scrolls before the async operation is complete).